charts/openstack-cluster/templates/_helpers.tpl

```diff
@@ -237,6 +228,13 @@ files:
     owner: root:root
     permissions: "0644"
 {{- end }}
+preKubeadmCommands:
+  - bash -c 'grep -q "[plugins."io.containerd.grpc.v1.cri".registry]" /etc/containerd/config.toml || cat << EOF >> /etc/containerd/config.toml
+
+[plugins."io.containerd.grpc.v1.cri".registry]
+  config_path = "/etc/containerd/certs.d"
+
+EOF'
```

Comment:
This needs to be inside the same conditional as the old `files` entry.

Also, did you test this yet? I'm almost certain this is not valid YAML or bash because of all the quotes...

I think this should work:

```yaml
preKubeadmCommands:
  - |
      bash -s <<EOF
      grep -q '[plugins."io.containerd.grpc.v1.cri".registry]' /etc/containerd/config.toml && exit
      cat <<CONTENT >> /etc/containerd/config.toml
      [plugins."io.containerd.grpc.v1.cri".registry]
      config_path = "/etc/containerd/certs.d"
      CONTENT
      EOF
```

Reply:
yea I realised it wasn't working after I pushed it - didn't get a chance to fix it until now - sorry about that